core/src/Platforms/iOS/EmbeddedWebview/EmbeddedWebUI.cs

```diff
@@ -65,29 +54,34 @@ public static void SetAuthorizationResult(AuthorizationResult authorizationResul
 
         public void Authenticate(Uri authorizationUri, Uri redirectUri, RequestContext requestContext)
         {
+            UIViewController vc = null;
+            InvokeOnMainThread(() =>
+            {
+                UIWindow window = UIApplication.SharedApplication.KeyWindow;
+                vc = CoreUIParent.FindCurrentViewController(window.RootViewController);
+            });
             try
             {
-                this.parameters.CallerViewController.InvokeOnMainThread(() =>
+                vc.InvokeOnMainThread(() =>
                 {
                     var navigationController =
                         new AuthenticationAgentUINavigationController(authorizationUri.AbsoluteUri,
-                            redirectUri.OriginalString, CallbackMethod, this.parameters.PreferredStatusBarStyle);
+                            redirectUri.OriginalString, CallbackMethod, CoreUIParent.PreferredStatusBarStyle);
 
-                    navigationController.ModalPresentationStyle = this.parameters.ModalPresentationStyle;
-                    navigationController.ModalTransitionStyle = this.parameters.ModalTransitionStyle;
-                    navigationController.TransitioningDelegate = this.parameters.TransitioningDelegate;
+                    navigationController.ModalPresentationStyle = CoreUIParent.ModalPresentationStyle;
+                    navigationController.ModalTransitionStyle = CoreUIParent.ModalTransitionStyle;
+                    navigationController.TransitioningDelegate = vc.TransitioningDelegate;
 
-                    this.parameters.CallerViewController.PresentViewController(navigationController, true, null);
+                    vc.PresentViewController(navigationController, true, null);
                 });
             }
             catch (Exception ex)
             {
-                this.parameters = null;
-                throw new AdalException(AdalError.AuthenticationUiFailed, ex);
+                throw new Client.MsalException(MsalError.AuthenticationUiFailed, ex.ToString());
```

Comment:
>hrow new Client.MsalException(MsalError.AuthenticationUiFailed, ex.ToString()); [](start = 17, length = 79)

this is in core. not sure why this is an msal exception?

Reply:
not as clear-cut as I thought it would be...going to revert changes and leave as-is...will make a task for this.